// test/stubs/AlphaVantageServiceStub.js
import sinon from 'sinon';
import StocksSymbols from 'app/service/StocksSymbols'
import AlphaVantageService from 'app/service/AlphaVantageService';

export default class AlphaVantageServiceStub
{
    static was_wrapped = false;
    static start() {
        if(AlphaVantageServiceStub.was_wrapped) {
            return;
        }
        AlphaVantageServiceStub.was_wrapped = true;

        let _info_helper_stub = sinon.stub(AlphaVantageService, '_info_helper',(symbol,callback)=>{
            if(symbol === StocksSymbols.MSFT) {
                callback(AlphaVantageServiceStub.fakeDate());
            } else if (symbol === StocksSymbols.APPL) {

            }

        });

        //_info_helper_stub.yields();
    }

    static linesFromDifferentDays()
    {
            return [{
                time: 1499952660,
                price: 71.48,
                symbol: 'MSFT',
                localTime: 1499952673.79,
                newYorkTime: '2017-07-13 09:31:00' },
                 {
                    time: 1499889600,
                    price: 71.15,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 16:00:00' },
                 {
                    time: 1499889540,
                    price: 71.19,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:59:00' },
                 {
                    time: 1499889480,
                    price: 71.1901,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:58:00' },
                 {
                    time: 1499889420,
                    price: 71.15,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:57:00' },
                 {
                    time: 1499889360,
                    price: 71.175,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:56:00' },
                 {
                    time: 1499889300,
                    price: 71.155,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:55:00' },
                 {
                    time: 1499889240,
                    price: 71.16,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:54:00' },
                 {
                    time: 1499889180,
                    price: 71.14,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:53:00' },
                 {
                    time: 1499889120,
                    price: 71.175,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:52:00' },
                 {
                    time: 1499889060,
                    price: 71.21,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:51:00' },
                 {
                    time: 1499889000,
                    price: 71.205,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:50:00' },
                 {
                    time: 1499888940,
                    price: 71.24,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:49:00' },
                 {
                    time: 1499888880,
                    price: 71.235,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:48:00' },
                 {
                    time: 1499888820,
                    price: 71.235,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:47:00' },
                 {
                    time: 1499888760,
                    price: 71.23,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:46:00' },
                 {
                    time: 1499888700,
                    price: 71.2,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:45:00' },
                 {
                    time: 1499888640,
                    price: 71.2,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:44:00' },
                 {
                    time: 1499888580,
                    price: 71.175,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:43:00' },
                 {
                    time: 1499888520,
                    price: 71.175,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:42:00' },
                 {
                    time: 1499888460,
                    price: 71.17,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:41:00' },
                 {
                    time: 1499888400,
                    price: 71.165,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:40:00' },
                 {
                    time: 1499888340,
                    price: 71.15,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:39:00' },
                 {
                    time: 1499888280,
                    price: 71.16,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:38:00' },
                 {
                    time: 1499888220,
                    price: 71.17,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:37:00' },
                 {
                    time: 1499888160,
                    price: 71.16,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:36:00' },
                 {
                    time: 1499888100,
                    price: 71.14,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:35:00' },
                 {
                    time: 1499888040,
                    price: 71.11,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:34:00' },
                 {
                    time: 1499887980,
                    price: 71.16,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:33:00' },
                 {
                    time: 1499887920,
                    price: 71.15,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:32:00' },
                 {
                    time: 1499887860,
                    price: 71.12,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:31:00' },
                 {
                    time: 1499887800,
                    price: 71.095,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:30:00' },
                 {
                    time: 1499887740,
                    price: 71.06,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:29:00' },
                 {
                    time: 1499887680,
                    price: 71.09,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:28:00' },
                 {
                    time: 1499887620,
                    price: 71.07,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:27:00' },
                 {
                    time: 1499887560,
                    price: 71.08,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:26:00' },
                 {
                    time: 1499887500,
                    price: 71.1,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:25:00' },
                 {
                    time: 1499887440,
                    price: 71.125,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:24:00' },
                 {
                    time: 1499887380,
                    price: 71.1239,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:23:00' },
                 {
                    time: 1499887320,
                    price: 71.115,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:22:00' },
                 {
                    time: 1499887260,
                    price: 71.115,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:21:00' },
                 {
                    time: 1499887200,
                    price: 71.1039,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:20:00' },
                 {
                    time: 1499887140,
                    price: 71.1015,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:19:00' },
                 {
                    time: 1499887080,
                    price: 71.12,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:18:00' },
                 {
                    time: 1499887020,
                    price: 71.115,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:17:00' },
                 {
                    time: 1499886960,
                    price: 71.12,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:16:00' },
                 {
                    time: 1499886900,
                    price: 71.11,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:15:00' },
                 {
                    time: 1499886840,
                    price: 71.11,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:14:00' },
                 {
                    time: 1499886780,
                    price: 71.1089,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:13:00' },
                 {
                    time: 1499886720,
                    price: 71.07,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:12:00' },
                 {
                    time: 1499886660,
                    price: 71.06,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:11:00' },
                 {
                    time: 1499886600,
                    price: 71.09,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:10:00' },
                 {
                    time: 1499886540,
                    price: 71.05,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:09:00' },
                 {
                    time: 1499886480,
                    price: 71.04,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:08:00' },
                 {
                    time: 1499886420,
                    price: 71.045,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:07:00' },
                 {
                    time: 1499886360,
                    price: 71.025,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:06:00' },
                 {
                    time: 1499886300,
                    price: 71,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:05:00' },
                 {
                    time: 1499886240,
                    price: 70.975,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:04:00' },
                 {
                    time: 1499886180,
                    price: 70.985,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:03:00' },
                 {
                    time: 1499886120,
                    price: 70.99,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:02:00' },
                 {
                    time: 1499886060,
                    price: 70.99,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:01:00' },
                 {
                    time: 1499886000,
                    price: 70.99,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 15:00:00' },
                 {
                    time: 1499885940,
                    price: 71.01,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:59:00' },
                 {
                    time: 1499885880,
                    price: 71,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:58:00' },
                 {
                    time: 1499885820,
                    price: 70.9948,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:57:00' },
                 {
                    time: 1499885760,
                    price: 70.99,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:56:00' },
                 {
                    time: 1499885700,
                    price: 70.99,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:55:00' },
                 {
                    time: 1499885640,
                    price: 71.009,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:54:00' },
                 {
                    time: 1499885580,
                    price: 70.9885,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:53:00' },
                 {
                    time: 1499885520,
                    price: 71,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:52:00' },
                 {
                    time: 1499885460,
                    price: 71.005,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:51:00' },
                 {
                    time: 1499885400,
                    price: 70.97,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:50:00' },
                 {
                    time: 1499885340,
                    price: 70.96,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:49:00' },
                 {
                    time: 1499885280,
                    price: 71,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:48:00' },
                 {
                    time: 1499885220,
                    price: 71.0038,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:47:00' },
                 {
                    time: 1499885160,
                    price: 70.995,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:46:00' },
                 {
                    time: 1499885100,
                    price: 70.991,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:45:00' },
                 {
                    time: 1499885040,
                    price: 71.04,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:44:00' },
                 {
                    time: 1499884980,
                    price: 71.01,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:43:00' },
                 {
                    time: 1499884920,
                    price: 71.025,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:42:00' },
                 {
                    time: 1499884860,
                    price: 70.9983,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:41:00' },
                 {
                    time: 1499884800,
                    price: 70.985,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:40:00' },
                 {
                    time: 1499884740,
                    price: 70.959,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:39:00' },
                 {
                    time: 1499884680,
                    price: 70.958,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:38:00' },
                 {
                    time: 1499884620,
                    price: 70.96,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:37:00' },
                 {
                    time: 1499884560,
                    price: 70.97,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:36:00' },
                 {
                    time: 1499884500,
                    price: 70.99,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:35:00' },
                 {
                    time: 1499884440,
                    price: 70.9865,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:34:00' },
                 {
                    time: 1499884380,
                    price: 70.9765,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:33:00' },
                 {
                    time: 1499884320,
                    price: 70.955,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:32:00' },
                 {
                    time: 1499884260,
                    price: 70.97,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:31:00' },
                 {
                    time: 1499884200,
                    price: 70.975,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:30:00' },
                 {
                    time: 1499884140,
                    price: 70.97,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:29:00' },
                 {
                    time: 1499884080,
                    price: 70.95,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:28:00' },
                 {
                    time: 1499884020,
                    price: 70.94,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:27:00' },
                 {
                    time: 1499883960,
                    price: 70.9234,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:26:00' },
                 {
                    time: 1499883900,
                    price: 70.935,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:25:00' },
                 {
                    time: 1499883840,
                    price: 70.9301,
                    symbol: 'MSFT',
                    localTime: 1499952673.79,
                    newYorkTime: '2017-07-12 14:24:00' },
                 {
                    time: 1499883780,
                    price: 70.94,
                    symbol: 'MSFT',
                    localTime: 1499952673.8,
                    newYorkTime: '2017-07-12 14:23:00' },
                 {
                    time: 1499883720,
                    price: 70.905,
                    symbol: 'MSFT',
                    localTime: 1499952673.8,
                    newYorkTime: '2017-07-12 14:22:00' } ]
    }
    static fakeDate()
    {
            return { 'Meta Data':
                { '1. Information': 'Intraday (1min) prices and volumes',
                    '2. Symbol': 'MSFT',
                    '3. Last Refreshed': '2017-06-23 16:00:00',
                    '4. Interval': '1min',
                    '5. Output Size': 'Compact',
                    '6. Time Zone': 'US/Eastern' },
                'Time Series (1min)':
                    { '2017-06-23 16:00:00':
                        { '1. open': '71.0850',
                            '2. high': '71.2500',
                            '3. low': '71.0800',
                            '4. close': '71.2200',
                            '5. volume': '6055858' },
                        '2017-06-23 15:59:00':
                            { '1. open': '71.0900',
                                '2. high': '71.0900',
                                '3. low': '71.0800',
                                '4. close': '71.0900',
                                '5. volume': '307169' },
                        '2017-06-23 15:58:00':
                            { '1. open': '71.0500',
                                '2. high': '71.0900',
                                '3. low': '71.0450',
                                '4. close': '71.0850',
                                '5. volume': '274417' },
                        '2017-06-23 15:57:00':
                            { '1. open': '71.0600',
                                '2. high': '71.0600',
                                '3. low': '71.0400',
                                '4. close': '71.0450',
                                '5. volume': '251978' },
                        '2017-06-23 15:56:00':
                            { '1. open': '71.0300',
                                '2. high': '71.0600',
                                '3. low': '71.0000',
                                '4. close': '71.0600',
                                '5. volume': '310239' },
                        '2017-06-23 15:55:00':
                            { '1. open': '71.0100',
                                '2. high': '71.0300',
                                '3. low': '71.0000',
                                '4. close': '71.0300',
                                '5. volume': '144379' },
                        '2017-06-23 15:54:00':
                            { '1. open': '71.0000',
                                '2. high': '71.0200',
                                '3. low': '70.9800',
                                '4. close': '71.0100',
                                '5. volume': '194724' },
                        '2017-06-23 15:53:00':
                            { '1. open': '70.9900',
                                '2. high': '71.0100',
                                '3. low': '70.9700',
                                '4. close': '70.9950',
                                '5. volume': '163236' },
                        '2017-06-23 15:52:00':
                            { '1. open': '71.0050',
                                '2. high': '71.0100',
                                '3. low': '70.9700',
                                '4. close': '70.9900',
                                '5. volume': '124509' },
                        '2017-06-23 15:51:00':
                            { '1. open': '70.9900',
                                '2. high': '71.0400',
                                '3. low': '70.9700',
                                '4. close': '71.0000',
                                '5. volume': '295094' },
                        '2017-06-23 15:50:00':
                            { '1. open': '70.9700',
                                '2. high': '71.0300',
                                '3. low': '70.9300',
                                '4. close': '71.0000',
                                '5. volume': '238309' },
                        '2017-06-23 15:49:00':
                            { '1. open': '70.9500',
                                '2. high': '70.9800',
                                '3. low': '70.9400',
                                '4. close': '70.9800',
                                '5. volume': '80492' },
                        '2017-06-23 15:48:00':
                            { '1. open': '70.9750',
                                '2. high': '70.9800',
                                '3. low': '70.9450',
                                '4. close': '70.9450',
                                '5. volume': '64801' },
                        '2017-06-23 15:47:00':
                            { '1. open': '70.9500',
                                '2. high': '70.9900',
                                '3. low': '70.9400',
                                '4. close': '70.9750',
                                '5. volume': '72495' },
                        '2017-06-23 15:46:00':
                            { '1. open': '70.9899',
                                '2. high': '70.9900',
                                '3. low': '70.9550',
                                '4. close': '70.9550',
                                '5. volume': '83563' },
                        '2017-06-23 15:45:00':
                            { '1. open': '71.0000',
                                '2. high': '71.0100',
                                '3. low': '70.9800',
                                '4. close': '70.9850',
                                '5. volume': '114417' },
                        '2017-06-23 15:44:00':
                            { '1. open': '70.9500',
                                '2. high': '71.0000',
                                '3. low': '70.9400',
                                '4. close': '70.9900',
                                '5. volume': '72261' },
                        '2017-06-23 15:43:00':
                            { '1. open': '70.9100',
                                '2. high': '70.9500',
                                '3. low': '70.8900',
                                '4. close': '70.9500',
                                '5. volume': '76764' },
                        '2017-06-23 15:42:00':
                            { '1. open': '70.9400',
                                '2. high': '70.9400',
                                '3. low': '70.9100',
                                '4. close': '70.9100',
                                '5. volume': '27055' },
                        '2017-06-23 15:41:00':
                            { '1. open': '70.8699',
                                '2. high': '70.9367',
                                '3. low': '70.8600',
                                '4. close': '70.9367',
                                '5. volume': '67707' },
                        '2017-06-23 15:40:00':
                            { '1. open': '70.9000',
                                '2. high': '70.9000',
                                '3. low': '70.8554',
                                '4. close': '70.8700',
                                '5. volume': '106276' },
                        '2017-06-23 15:39:00':
                            { '1. open': '70.9500',
                                '2. high': '70.9500',
                                '3. low': '70.8800',
                                '4. close': '70.9000',
                                '5. volume': '63050' },
                        '2017-06-23 15:38:00':
                            { '1. open': '70.9550',
                                '2. high': '70.9600',
                                '3. low': '70.9300',
                                '4. close': '70.9400',
                                '5. volume': '48902' },
                        '2017-06-23 15:37:00':
                            { '1. open': '70.9950',
                                '2. high': '70.9950',
                                '3. low': '70.9400',
                                '4. close': '70.9500',
                                '5. volume': '46346' },
                        '2017-06-23 15:36:00':
                            { '1. open': '71.0063',
                                '2. high': '71.0100',
                                '3. low': '70.9700',
                                '4. close': '70.9950',
                                '5. volume': '53358' },
                        '2017-06-23 15:35:00':
                            { '1. open': '70.9650',
                                '2. high': '71.0150',
                                '3. low': '70.9500',
                                '4. close': '71.0050',
                                '5. volume': '82601' },
                        '2017-06-23 15:34:00':
                            { '1. open': '70.9400',
                                '2. high': '70.9700',
                                '3. low': '70.9350',
                                '4. close': '70.9650',
                                '5. volume': '32096' },
                        '2017-06-23 15:33:00':
                            { '1. open': '70.9100',
                                '2. high': '70.9400',
                                '3. low': '70.9000',
                                '4. close': '70.9400',
                                '5. volume': '26163' },
                        '2017-06-23 15:32:00':
                            { '1. open': '70.8850',
                                '2. high': '70.9100',
                                '3. low': '70.8800',
                                '4. close': '70.9000',
                                '5. volume': '29546' },
                        '2017-06-23 15:31:00':
                            { '1. open': '70.8600',
                                '2. high': '70.8900',
                                '3. low': '70.8500',
                                '4. close': '70.8900',
                                '5. volume': '24459' },
                        '2017-06-23 15:30:00':
                            { '1. open': '70.8700',
                                '2. high': '70.8700',
                                '3. low': '70.8500',
                                '4. close': '70.8500',
                                '5. volume': '34445' },
                        '2017-06-23 15:29:00':
                            { '1. open': '70.8900',
                                '2. high': '70.8900',
                                '3. low': '70.8350',
                                '4. close': '70.8600',
                                '5. volume': '89821' },
                        '2017-06-23 15:28:00':
                            { '1. open': '70.9000',
                                '2. high': '70.9100',
                                '3. low': '70.8850',
                                '4. close': '70.8850',
                                '5. volume': '35226' },
                        '2017-06-23 15:27:00':
                            { '1. open': '70.9250',
                                '2. high': '70.9250',
                                '3. low': '70.8900',
                                '4. close': '70.9000',
                                '5. volume': '22079' },
                        '2017-06-23 15:26:00':
                            { '1. open': '70.9200',
                                '2. high': '70.9300',
                                '3. low': '70.9100',
                                '4. close': '70.9200',
                                '5. volume': '37210' },
                        '2017-06-23 15:25:00':
                            { '1. open': '70.9200',
                                '2. high': '70.9500',
                                '3. low': '70.9100',
                                '4. close': '70.9300',
                                '5. volume': '47226' },
                        '2017-06-23 15:24:00':
                            { '1. open': '71.0000',
                                '2. high': '71.0100',
                                '3. low': '70.9300',
                                '4. close': '70.9300',
                                '5. volume': '48434' },
                        '2017-06-23 15:23:00':
                            { '1. open': '70.9750',
                                '2. high': '71.0200',
                                '3. low': '70.9700',
                                '4. close': '71.0000',
                                '5. volume': '176386' },
                        '2017-06-23 15:22:00':
                            { '1. open': '70.9250',
                                '2. high': '70.9900',
                                '3. low': '70.9100',
                                '4. close': '70.9750',
                                '5. volume': '107537' },
                        '2017-06-23 15:21:00':
                            { '1. open': '70.9200',
                                '2. high': '70.9599',
                                '3. low': '70.9200',
                                '4. close': '70.9200',
                                '5. volume': '91062' },
                        '2017-06-23 15:20:00':
                            { '1. open': '70.8710',
                                '2. high': '70.9400',
                                '3. low': '70.8700',
                                '4. close': '70.9200',
                                '5. volume': '129233' },
                        '2017-06-23 15:19:00':
                            { '1. open': '70.8900',
                                '2. high': '70.8900',
                                '3. low': '70.8600',
                                '4. close': '70.8700',
                                '5. volume': '29117' },
                        '2017-06-23 15:18:00':
                            { '1. open': '70.8600',
                                '2. high': '70.9050',
                                '3. low': '70.8600',
                                '4. close': '70.8950',
                                '5. volume': '52363' },
                        '2017-06-23 15:17:00':
                            { '1. open': '70.8450',
                                '2. high': '70.8662',
                                '3. low': '70.8400',
                                '4. close': '70.8600',
                                '5. volume': '17632' },
                        '2017-06-23 15:16:00':
                            { '1. open': '70.8600',
                                '2. high': '70.8600',
                                '3. low': '70.8200',
                                '4. close': '70.8450',
                                '5. volume': '53349' },
                        '2017-06-23 15:15:00':
                            { '1. open': '70.9000',
                                '2. high': '70.9000',
                                '3. low': '70.8500',
                                '4. close': '70.8600',
                                '5. volume': '52193' },
                        '2017-06-23 15:14:00':
                            { '1. open': '70.9050',
                                '2. high': '70.9100',
                                '3. low': '70.8900',
                                '4. close': '70.8950',
                                '5. volume': '42796' },
                        '2017-06-23 15:13:00':
                            { '1. open': '70.8800',
                                '2. high': '70.9100',
                                '3. low': '70.8800',
                                '4. close': '70.9000',
                                '5. volume': '28096' },
                        '2017-06-23 15:12:00':
                            { '1. open': '70.8850',
                                '2. high': '70.9100',
                                '3. low': '70.8800',
                                '4. close': '70.8850',
                                '5. volume': '49364' },
                        '2017-06-23 15:11:00':
                            { '1. open': '70.8650',
                                '2. high': '70.8900',
                                '3. low': '70.8650',
                                '4. close': '70.8800',
                                '5. volume': '28015' },
                        '2017-06-23 15:10:00':
                            { '1. open': '70.8650',
                                '2. high': '70.8700',
                                '3. low': '70.8550',
                                '4. close': '70.8600',
                                '5. volume': '19477' },
                        '2017-06-23 15:09:00':
                            { '1. open': '70.8600',
                                '2. high': '70.8800',
                                '3. low': '70.8550',
                                '4. close': '70.8601',
                                '5. volume': '35287' },
                        '2017-06-23 15:08:00':
                            { '1. open': '70.8461',
                                '2. high': '70.8600',
                                '3. low': '70.8450',
                                '4. close': '70.8600',
                                '5. volume': '16653' },
                        '2017-06-23 15:07:00':
                            { '1. open': '70.8189',
                                '2. high': '70.8500',
                                '3. low': '70.8150',
                                '4. close': '70.8500',
                                '5. volume': '41611' },
                        '2017-06-23 15:06:00':
                            { '1. open': '70.8100',
                                '2. high': '70.8100',
                                '3. low': '70.7900',
                                '4. close': '70.8100',
                                '5. volume': '12900' },
                        '2017-06-23 15:05:00':
                            { '1. open': '70.7956',
                                '2. high': '70.8200',
                                '3. low': '70.7900',
                                '4. close': '70.8100',
                                '5. volume': '31138' },
                        '2017-06-23 15:04:00':
                            { '1. open': '70.8000',
                                '2. high': '70.8100',
                                '3. low': '70.7800',
                                '4. close': '70.7950',
                                '5. volume': '27141' },
                        '2017-06-23 15:03:00':
                            { '1. open': '70.8600',
                                '2. high': '70.8600',
                                '3. low': '70.7800',
                                '4. close': '70.7900',
                                '5. volume': '85447' },
                        '2017-06-23 15:02:00':
                            { '1. open': '70.9200',
                                '2. high': '70.9200',
                                '3. low': '70.8610',
                                '4. close': '70.8610',
                                '5. volume': '53062' },
                        '2017-06-23 15:01:00':
                            { '1. open': '70.9400',
                                '2. high': '70.9500',
                                '3. low': '70.9150',
                                '4. close': '70.9262',
                                '5. volume': '29650' },
                        '2017-06-23 15:00:00':
                            { '1. open': '70.9450',
                                '2. high': '70.9700',
                                '3. low': '70.9400',
                                '4. close': '70.9500',
                                '5. volume': '30682' },
                        '2017-06-23 14:59:00':
                            { '1. open': '70.9800',
                                '2. high': '70.9850',
                                '3. low': '70.9500',
                                '4. close': '70.9500',
                                '5. volume': '33974' },
                        '2017-06-23 14:58:00':
                            { '1. open': '71.0050',
                                '2. high': '71.0100',
                                '3. low': '70.9700',
                                '4. close': '70.9800',
                                '5. volume': '28817' },
                        '2017-06-23 14:57:00':
                            { '1. open': '71.0100',
                                '2. high': '71.0100',
                                '3. low': '71.0000',
                                '4. close': '71.0100',
                                '5. volume': '8136' },
                        '2017-06-23 14:56:00':
                            { '1. open': '70.9800',
                                '2. high': '71.0100',
                                '3. low': '70.9800',
                                '4. close': '71.0050',
                                '5. volume': '36869' },
                        '2017-06-23 14:55:00':
                            { '1. open': '70.9900',
                                '2. high': '70.9950',
                                '3. low': '70.9800',
                                '4. close': '70.9881',
                                '5. volume': '26018' },
                        '2017-06-23 14:54:00':
                            { '1. open': '70.9800',
                                '2. high': '70.9900',
                                '3. low': '70.9700',
                                '4. close': '70.9900',
                                '5. volume': '8857' },
                        '2017-06-23 14:53:00':
                            { '1. open': '70.9750',
                                '2. high': '70.9800',
                                '3. low': '70.9600',
                                '4. close': '70.9800',
                                '5. volume': '10345' },
                        '2017-06-23 14:52:00':
                            { '1. open': '70.9700',
                                '2. high': '70.9800',
                                '3. low': '70.9600',
                                '4. close': '70.9750',
                                '5. volume': '14041' },
                        '2017-06-23 14:51:00':
                            { '1. open': '70.9550',
                                '2. high': '70.9700',
                                '3. low': '70.9500',
                                '4. close': '70.9700',
                                '5. volume': '10448' },
                        '2017-06-23 14:50:00':
                            { '1. open': '70.9700',
                                '2. high': '70.9700',
                                '3. low': '70.9400',
                                '4. close': '70.9500',
                                '5. volume': '46219' },
                        '2017-06-23 14:49:00':
                            { '1. open': '70.9800',
                                '2. high': '70.9800',
                                '3. low': '70.9600',
                                '4. close': '70.9600',
                                '5. volume': '14305' },
                        '2017-06-23 14:48:00':
                            { '1. open': '70.9799',
                                '2. high': '70.9900',
                                '3. low': '70.9700',
                                '4. close': '70.9750',
                                '5. volume': '19476' },
                        '2017-06-23 14:47:00':
                            { '1. open': '70.9554',
                                '2. high': '70.9800',
                                '3. low': '70.9500',
                                '4. close': '70.9720',
                                '5. volume': '13219' },
                        '2017-06-23 14:46:00':
                            { '1. open': '70.9600',
                                '2. high': '70.9700',
                                '3. low': '70.9600',
                                '4. close': '70.9600',
                                '5. volume': '12764' },
                        '2017-06-23 14:45:00':
                            { '1. open': '71.0000',
                                '2. high': '71.0000',
                                '3. low': '70.9600',
                                '4. close': '70.9600',
                                '5. volume': '16014' },
                        '2017-06-23 14:44:00':
                            { '1. open': '71.0600',
                                '2. high': '71.0600',
                                '3. low': '71.0000',
                                '4. close': '71.0000',
                                '5. volume': '89410' },
                        '2017-06-23 14:43:00':
                            { '1. open': '71.0250',
                                '2. high': '71.0600',
                                '3. low': '71.0200',
                                '4. close': '71.0600',
                                '5. volume': '84439' },
                        '2017-06-23 14:42:00':
                            { '1. open': '71.0158',
                                '2. high': '71.0400',
                                '3. low': '70.9850',
                                '4. close': '71.0300',
                                '5. volume': '155124' },
                        '2017-06-23 14:41:00':
                            { '1. open': '71.0100',
                                '2. high': '71.0350',
                                '3. low': '70.9750',
                                '4. close': '71.0100',
                                '5. volume': '108395' },
                        '2017-06-23 14:40:00':
                            { '1. open': '71.0500',
                                '2. high': '71.0500',
                                '3. low': '71.0000',
                                '4. close': '71.0001',
                                '5. volume': '29344' },
                        '2017-06-23 14:39:00':
                            { '1. open': '71.0301',
                                '2. high': '71.0500',
                                '3. low': '71.0100',
                                '4. close': '71.0500',
                                '5. volume': '40597' },
                        '2017-06-23 14:38:00':
                            { '1. open': '71.0300',
                                '2. high': '71.0500',
                                '3. low': '71.0050',
                                '4. close': '71.0400',
                                '5. volume': '62052' },
                        '2017-06-23 14:37:00':
                            { '1. open': '71.0400',
                                '2. high': '71.0500',
                                '3. low': '71.0000',
                                '4. close': '71.0300',
                                '5. volume': '67567' },
                        '2017-06-23 14:36:00':
                            { '1. open': '71.0200',
                                '2. high': '71.0500',
                                '3. low': '70.9900',
                                '4. close': '71.0500',
                                '5. volume': '49053' },
                        '2017-06-23 14:35:00':
                            { '1. open': '71.0000',
                                '2. high': '71.0550',
                                '3. low': '71.0000',
                                '4. close': '71.0300',
                                '5. volume': '61763' },
                        '2017-06-23 14:34:00':
                            { '1. open': '70.9700',
                                '2. high': '71.0283',
                                '3. low': '70.9700',
                                '4. close': '71.0090',
                                '5. volume': '49543' },
                        '2017-06-23 14:33:00':
                            { '1. open': '70.9545',
                                '2. high': '70.9858',
                                '3. low': '70.9500',
                                '4. close': '70.9800',
                                '5. volume': '20459' },
                        '2017-06-23 14:32:00':
                            { '1. open': '70.9700',
                                '2. high': '70.9800',
                                '3. low': '70.9500',
                                '4. close': '70.9500',
                                '5. volume': '18114' },
                        '2017-06-23 14:31:00':
                            { '1. open': '70.9800',
                                '2. high': '70.9900',
                                '3. low': '70.9500',
                                '4. close': '70.9600',
                                '5. volume': '38967' },
                        '2017-06-23 14:30:00':
                            { '1. open': '71.0150',
                                '2. high': '71.0150',
                                '3. low': '70.9800',
                                '4. close': '70.9801',
                                '5. volume': '20995' },
                        '2017-06-23 14:29:00':
                            { '1. open': '71.0300',
                                '2. high': '71.0400',
                                '3. low': '71.0000',
                                '4. close': '71.0100',
                                '5. volume': '37740' },
                        '2017-06-23 14:28:00':
                            { '1. open': '71.0500',
                                '2. high': '71.0550',
                                '3. low': '71.0200',
                                '4. close': '71.0342',
                                '5. volume': '25967' },
                        '2017-06-23 14:27:00':
                            { '1. open': '71.0300',
                                '2. high': '71.0600',
                                '3. low': '71.0000',
                                '4. close': '71.0600',
                                '5. volume': '83133' },
                        '2017-06-23 14:26:00':
                            { '1. open': '71.0300',
                                '2. high': '71.0350',
                                '3. low': '71.0000',
                                '4. close': '71.0350',
                                '5. volume': '48871' },
                        '2017-06-23 14:25:00':
                            { '1. open': '71.0000',
                                '2. high': '71.0301',
                                '3. low': '70.9900',
                                '4. close': '71.0301',
                                '5. volume': '49864' },
                        '2017-06-23 14:24:00':
                            { '1. open': '70.9900',
                                '2. high': '71.0250',
                                '3. low': '70.9900',
                                '4. close': '71.0050',
                                '5. volume': '38800' },
                        '2017-06-23 14:23:00':
                            { '1. open': '71.0143',
                                '2. high': '71.0200',
                                '3. low': '70.9700',
                                '4. close': '71.0000',
                                '5. volume': '60862' },
                        '2017-06-23 14:22:00':
                            { '1. open': '71.0050',
                                '2. high': '71.0300',
                                '3. low': '70.9900',
                                '4. close': '71.0100',
                                '5. volume': '25003' },
                        '2017-06-23 14:21:00':
                            { '1. open': '70.9750',
                                '2. high': '71.0100',
                                '3. low': '70.9700',
                                '4. close': '71.0000',
                                '5. volume': '65993' } } }
    }
}
